### ISSUE TITLE
isa(x,Number)

### ISSUE THREAD

**Issue by sswatson** [symengine/SymEngine.jl#82]
It seems undesirable that a variable of type `SymEngine.Basic` is a `Number` but throws an error when coerced to that type. The type `SymPy.Sym` does not have this issue. 

```julia
julia> using SymEngine
julia> @vars x
```

```julia
julia> isa(x,Number)
true
```

```julia
julia> Number[1,2,3,x]
ERROR: ArgumentError: Object can have no free symbols
 in N(::SymEngine.BasicType{Val{:Symbol}}) at /Users/sswatson/.julia/v0.5/SymEngine/src/numerics.jl:122
 in setindex!(::Array{Number,1}, ::SymEngine.Basic, ::Int64) at ./array.jl:415
 in getindex(::Type{Number}, ::Int64, ::Int64, ::Int64, ::SymEngine.Basic, ::Vararg{SymEngine.Basic,N}) at ./array.jl:132
```

```julia 
julia> using SymPy; @syms y
(y,)

julia> Number[1,2,3,y]
4-element Array{Number,1}:
 1
 2
 3
 y
```

**Comment by ChrisRackauckas** [comment]
Yeah, that's peculiar. I never noticed this. It should definitely be allowed in an array of numbers.

**Comment by isuruf** [comment]
Fix in #83. Added the example here as a testcase.
